From f39d6324c0a599f831268835f53414892aa0a9b2 Mon Sep 17 00:00:00 2001
From: cfigueir <carlos.figueiredofelipe@windriver.com>
Date: Tue, 30 Jun 2026 11:54:41 -0300
Subject: [PATCH] Add Prometheus support to Aodh

This patch adds the Aodh prometheus configuration structure to be set
in the respective Helm Chart overrides.

Configuration keys are under the prefix `.Values.conf.prometheus.`.
Following is a Charts configuration example:
`
conf:
  prometheus:
    host: prometheus.monitoring.svc.cluster.local
    port: 9090
`

An `aodh_notifier` override key is added with notifier specific
configuration outside of the aodh.conf rendering structure.

The `aodh_notifier.tls` allow users to configure a custom host cert
path to be used by the rest notifier when validating the host identity
when performing alarm notification HTTPs requests.

Signed-off-by: cfigueir <carlos.figueiredofelipe@windriver.com>
[Fix Aodh cron-job-alarms-cleaner volume reference]
Signed-off-by: cfigueir <carlos.figueiredofelipe@windriver.com>
[Support for Aodh rest notifier host certificate]
Signed-off-by: cfigueir <carlos.figueiredofelipe@windriver.com>
---
 aodh/templates/cron-job-alarms-cleaner.yaml | 2 +-
 aodh/templates/deployment-evaluator.yaml    | 5 +++++
 aodh/values.yaml                            | 7 +++++++
 3 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/aodh/templates/cron-job-alarms-cleaner.yaml b/aodh/templates/cron-job-alarms-cleaner.yaml
index aeb3b9356..dfeef549d 100644
--- a/aodh/templates/cron-job-alarms-cleaner.yaml
+++ b/aodh/templates/cron-job-alarms-cleaner.yaml
@@ -96,7 +96,7 @@ spec:
               secret:
                 secretName: aodh-etc
                 defaultMode: 0444
-            - name: aodh-et-etc-snippets
+            - name: aodh-etc-snippets
 {{- if $etcSources }}
               projected:
                 sources:
diff --git a/aodh/templates/deployment-evaluator.yaml b/aodh/templates/deployment-evaluator.yaml
index 662063880..17e70e088 100644
--- a/aodh/templates/deployment-evaluator.yaml
+++ b/aodh/templates/deployment-evaluator.yaml
@@ -76,6 +76,11 @@ spec:
                 command:
                   - /tmp/aodh-evaluator.sh
                   - stop
+          env:
+            - name: PROMETHEUS_HOST
+              value: {{ .Values.conf.prometheus.host | quote }}
+            - name: PROMETHEUS_PORT
+              value: {{ .Values.conf.prometheus.port | quote }}
           volumeMounts:
             - name: pod-tmp
               mountPath: /tmp
diff --git a/aodh/values.yaml b/aodh/values.yaml
index 12489bb0a..8f3716514 100644
--- a/aodh/values.yaml
+++ b/aodh/values.yaml
@@ -548,6 +548,13 @@ conf:
       class: oslo_log.formatters.ContextFormatter
     formatter_default:
       format: "%(message)s"
+  prometheus:
+    host: null
+    port: null
+
+aodh_notifier:
+  tls:
+    host_cert: "/var/opt/openstack/certs/aodh-rest-notifier.pem"
 
 secrets:
   identity:
-- 
2.43.0

